x-pack/plugins/ecs_data_quality_dashboard/server/routes/get_index_stats.ts

```diff
@@ -57,24 +65,26 @@ export const getIndexStatsRoute = (router: IRouter, logger: Logger) => {
           if (startDate && endDate) {
             const decodedStartDate = decodeURIComponent(startDate);
             const decodedEndDate = decodeURIComponent(endDate);
+            const meteringStats = await fetchMeteringStats(
+              client,
+              decodedIndexName,
+              request.headers.authorization
+            );
+
+            const meteringStatsIndices = parseMeteringStats(meteringStats.indices);
 
-            const indices = await fetchAvailableIndices(esClient, {
+            const availableIndices = await fetchAvailableIndices(esClient, {
               indexPattern: decodedIndexName,
               startDate: decodedStartDate,
               endDate: decodedEndDate,
             });
-            const availableIndices = indices?.aggregations?.index?.buckets?.reduce(
-              (acc: Record<string, IndicesStatsIndicesStats>, { key }: { key: string }) => {
-                if (stats.indices?.[key]) {
-                  acc[key] = stats.indices?.[key];
-                }
-                return acc;
-              },
-              {}
+            const indices = pickAvailableMeteringStats(
+              availableIndices?.aggregations?.index?.buckets,
+              meteringStatsIndices
```

Comment:
Could we return with an error or an empty array when we don't have `availableIndices` buckets? The same for `meteringStatsIndices` and even `meteringStats.indices` (this last one I am not sure if it can be undefined, types say it can not, but we have `?` in the implementation, one or the other is wrong).

These undefined values are dragged down, forcing `pickAvailableMeteringStats` and `parseMeteringStats` to work with `undefined` parameters, which does not make much sense, they shouldn't even be called if we don't have the data.

Reply:
Thank you @semd , good point! I've added checks to see if have indices returned from `availableIndices ` and `meteringStats.indices` , so it returns an empty object straight away and log the warning message. https://github.com/elastic/kibana/pull/179666/commits/b64973f917fea8d8cd18e4523ebe29caa9f9091f